### ISSUE TITLE
Phase 2: Remove legacy FPN code from Guardian

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/mozilla-vpn-client#6568]
┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/VPN-4530)
┆Reporter: Santiago Andrigo

**Comment by data-sync-user** [comment]
➤ Sarah Bird commented:

Jeremy Swinarton Am thinking we probably to do data deletion in a second step.

**Comment by data-sync-user** [comment]
➤ Santiago Andrigo commented:

Sarah Bird just checking - you won’t be around for Phase 2 (June 20th), so are you thinking that you can just create the PR during Grey Goose and then Jeremy can review and push it live in June itself?

**Comment by data-sync-user** [comment]
➤ Sarah Bird commented:

Santiago Andrigo yes.

**Comment by data-sync-user** [comment]
➤ Sarah Bird commented:

Latest on this. We have a PR. Everything works fine locally and all tests pass. Unfortunately when I deployed to staging (v1.30.b1) it broke guardian. Specifically, authentication broke. 

I reverted the change on staging (v1.30.b2) and opened a new PR. https://github.com/mozilla-services/guardian-website/pull/1632 ( https://github.com/mozilla-services/guardian-website/pull/1632|smart-link )  Jeremy Swinarton's going to have to take it from here.

**Comment by data-sync-user** [comment]
➤ Jeremy Swinarton commented:

Moving this to blocked for now, because we have a dot release up on stage at the moment and I’m unable to test this there.

I’ve tried to reproduce locally and on the Heroku dev server with no luck so far.

**Comment by data-sync-user** [comment]
➤ Jeremy Swinarton commented:

(cc Sarah Bird not because I have anything I need you to do, just because I think you will find this as interesting as I did)

Ok, I figured this out. Looking at the pattern of requests on stage, I discovered that the FxA callback URL for Oauth has a domain of stage.guardian.nonprod.cloudops.mozgcp.net , NOT stage-vpn.guardian.nonprod.cloudops.mozgcp.net . Note the -vpn in the first subdomain. The former is the staging domain for FPN, the latter for VPN. Apparently, up to this point, FPN was catching this request and redirecting it to the VPN site and all proceeded as normal. With the FPN deletion this redirect was broken.

What is super interesting to me about this is that because we removed the distinction between the sites, what is on FPN stage is now identical to what is on VPN stage. So the redirect almost worked, but the session cookie didn’t get passed from one domain to the next. Therefore, on the success callback, the session was blank, and was missing the crucial state token that we validate in the success step. Thus the state_token_inequality error.

FASCINATING. 

Anyway, the appropriate fix to this is to have FxA update their client config which I will coordinate when I am in next week.

**Comment by data-sync-user** [comment]
➤ Lesley Norton commented:

Apparently, up to this point, FPN was catching this request and redirecting it to the VPN site and all proceeded as normal. With the FPN deletion this redirect was broken.

I may have to print this out and have it framed.

**Comment by data-sync-user** [comment]
➤ Jeremy Swinarton commented:

Found the culprit 🙂 https://github.com/mozilla-services/guardian-website/pull/1632/files#diff-51d63df73dde8ff810280c9b9e429aa88a09bb6d22e21bef49ead90e77210ecbL41-L44 ( https://github.com/mozilla-services/guardian-website/pull/1632/files#diff-51d63df73dde8ff810280c9b9e429aa88a09bb6d22e21bef49ead90e77210ecbL41-L44 )

**Comment by data-sync-user** [comment]
➤ Sarah Bird commented:

cc Bianca Danforth fyi on the sadness of relying parties (this is an unusual one though).

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Jeremy Swinarton Will this go out before FPN Phase 2 or together with VPN 2.16? Should we schedule a regression on 1.30 Guardian version?

**Comment by data-sync-user** [comment]
➤ Jeremy Swinarton commented:

Hi Valentina Virlics , I will add a ticket for a 1.30 regression test. This is not tied to 2.16 and does not have to go out until after phase 2 is complete. Our current plan is to release this one week after the sunset date, which is June 27.

**Comment by data-sync-user** [comment]
➤ Bianca Hidecuti commented:

This ticket is verified thorugh https://mozilla-hub.atlassian.net/browse/QA-1958 ( https://mozilla-hub.atlassian.net/browse/QA-1958|smart-link ). Marking this verified as fixed.
